### ISSUE TITLE
Image Repository requires basic auth

### ISSUE THREAD

**Issue by ohuarte-matrix** [aws/eks-charts#443]
**Describe the bug**
The repository 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon/aws-load-balancer-controller requires basic auth, so if you try to install using this value it will fail with imagepullbackoff error in kubernetes

**Steps to reproduce**
helm repo add eks https://aws.github.io/eks-charts
helm install aws-load-balancer-controller eks/aws-load-balancer-controller --set clusterName=my-cluster
**Expected outcome**
pod created

**Environment**
 aws t2.medium with ubuntu
* Chart name: eks/aws-load-balancer-controller
* Chart version: 1.1.3
* Kubernetes version: 1.19
* Using EKS (yes/no), if so version?

**Additional Context**:

**Comment by privomark** [comment]
Not sure if it's helpful, but I had the same issue. Turns out we had an SCP explicitly denying `us-west-2` the region where AWS has their ECR and container for this chart.

If you run...

`aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 602401143452.dkr.ecr.us-west-2.amazonaws.com` 

...on your node that hosts the pod giving the error, it may provide more insight on your issue.

**Comment by ohuarte-matrix** [comment]
> Not sure if it's helpful, but I had the same issue. Turns out we had an SCP explicitly denying `us-west-2` the region where AWS has their ECR and container for this chart.
> 
> If you run...
> 
> `aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 602401143452.dkr.ecr.us-west-2.amazonaws.com`
> 
> ...on your node that hosts the pod giving the error, it may provide more insight on your issue.

Thanks for your help.

My aws account is created in  eu-west-2 - and it has some restrictions to use different zones, maybe you are right

I have tried to use your command but with eu-west-2 zone in aws ecr command.

```
aws ecr get-login-password --region eu-west-2 --profile saml | docker login --username AWS --password-stdin 602401143452.dkr.ecr.us-west-2.amazonaws.com

Error response from daemon: login attempt to https://602401143452.dkr.ecr.us-west-2.amazonaws.com/v2/ failed with status: 400 Bad Request
```


If I try to use with us-west-2 in aws command:
```
aws ecr get-login-password --region us-west-2 --profile saml | docker login --username AWS --password-stdin 602401143452.dkr.ecr.us-west-2.amazonaws.com 

An error occurred (AccessDeniedException) when calling the GetAuthorizationToken operation: User: arn:aws:sts:xxxxxxxx:assumed-role/XXXXAdministratorRole/ is not authorized to perform: ecr:GetAuthorizationToken on resource: * with an explicit deny
```

In any case, I have tried to change repository to  **docker.io/amazon/aws-alb-ingress-controller** and it works ok, is deployed without any problem, but the question is:
is   602401143452.dkr.ecr.us-west-2.amazonaws.com/ image  the same than **docker.io** ? Apparently I am using the same image tag version.

Thanks a lot for your help

**Comment by privomark** [comment]
Yea I think that's definitely your issue. The `is not authorized to perform: ecr:GetAuthorizationToken on resource: * with an explicit deny` is the exact message I got until we modified our SCP. 

Also, if you're running this on the EC2 Worker Node, you shouldn't need to pass the `--profile saml` 

As far as the "docker.io" image being the same as the one hosted on "602401143452.dkr.ecr.us-west-2.amazonaws.com", I have the same concerns as you do, which is why we allowed access to "us-west-2" instead of me referring to docker or even creating a local copy of the container and hosting it privately. Also, you never know when they update it with recent bug fixes.

Good luck!
